File: leetcode021.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        h = head = ListNode(-1)
        while l1 != None and l2 != None:
            if l1.val > l2.val:
                h.next = l2
                h = h.next
                l2 = l2.next
            else:
                h.next = l1
                h = h.next
                l1 = l1.next
        if l1 !=None:
            h.next = l1
        if l2 != None:
            h.next = l2
        return head.next

File: leetcodec++/leetcode021.cpp
//
// Created by 黄敏辉 on 2019-04-30.
//

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* head = NULL;
        ListNode* p = NULL;
        while (l1!= NULL && l2!=NULL){
            printf("%d, %d\n", l1->val, l2->val);
            if (l1->val < l2->val) {
                if (head == NULL){
                    head = p = l1;
                    l1 = l1->next;
                }else{
                    p->next = l1;
                    p = p->next;
                    l1 = l1->next;
                }

            }
            else {
                if (head == NULL){
                    head = p = l2;
                    l2 = l2->next;
                }else{
                    p->next = l2;
                    p = p->next;
                    l2 = l2->next;
                }

            }
        }
        if (l1){
            if (head == NULL)
                head = l1;
            else
                p->next = l1;
        }
        if (l2){
            if (head == NULL)
                head = l2;
            else
                p->next = l2;
        }
        return head;
    }
};